presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction.java

```diff
@@ -17,28 +17,88 @@
+        checkCondition(step > 0 ? stop >= start : stop < start, INVALID_FUNCTION_ARGUMENT,
+                "sequence stop value should be greater than or equal to start value if step is greater than zero otherwise stop should be less than start");
+
+        int length = Ints.checkedCast((stop - start) / step + 1L);
+        checkCondition(length > 0, INVALID_FUNCTION_ARGUMENT, "length of sequence is 0");
+
+        BlockBuilder blockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), length);
+        for (long i = 0, value = start; i < length; ++i, value += step) {
+            type.writeLong(blockBuilder, value);
+        }
+        return blockBuilder.build();
+    }
+
     @Description("Sequence function to generate synthetic arrays")
-    @ScalarFunction
+    @ScalarFunction("sequence")
+    @SqlType("array(bigint)")
+    public static Block sequence(@SqlType(StandardTypes.BIGINT) long start,
+                                 @SqlType(StandardTypes.BIGINT) long stop,
+                                 @SqlType(StandardTypes.BIGINT) long step)
+    {
+        return fixedWidthSequence(start, stop, step, BIGINT);
+    }
+
+    @ScalarFunction("sequence")
     @SqlType("array(bigint)")
-    public static Block sequence(@SqlType("bigint") long start, @SqlType("bigint") long end)
+    public static Block sequenceDefaultStep(@SqlType(StandardTypes.BIGINT) long start,
+                                            @SqlType(StandardTypes.BIGINT) long stop)
     {
-        checkCondition(end >= start, INVALID_FUNCTION_ARGUMENT, "sequence end value should be greater than or equal to start value");
-        int length = Ints.checkedCast(end - start + 1);
+        return fixedWidthSequence(start, stop, stop >= start ? 1 : -1, BIGINT);
+    }
+
+    @ScalarFunction("sequence")
+    @SqlType("array(timestamp)")
+    public static Block sequenceTimestampDayToSecond(@SqlType(StandardTypes.TIMESTAMP) long start,
+                                                     @SqlType(StandardTypes.TIMESTAMP) long end,
+                                                     @SqlType(StandardTypes.INTERVAL_DAY_TO_SECOND) long step)
+    {
+        return fixedWidthSequence(start, end, step, TIMESTAMP);
+    }
+
+    @ScalarFunction("sequence")
+    @SqlType("array(timestamp)")
+    public static Block sequenceTimestampYearToMonth(@SqlType(StandardTypes.TIMESTAMP) long start,
+                                                     @SqlType(StandardTypes.TIMESTAMP) long end,
+                                                     @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long step)
+    {
+        checkCondition(step != 0, INVALID_FUNCTION_ARGUMENT, "interval must not be zero");
+        checkCondition(step > 0 ? end >= start : end <= start, INVALID_FUNCTION_ARGUMENT,
+                "sequence end value should be greater than or equal to start value if step is greater than zero otherwise end should be less than start");
+
```

Comment:
> There also seem to be a bunch of unused functions in DateTimeOperators. Is that actually the case or are they being used and Intellij just can't tell where?

Those are introspected and registered dynamically by the server, so they don't have any direct callers that intellij can analyze.

What we do is we tell Intellij to suppress that inspection for methods annotated with `@ScalarOperator`